<a href="https://colab.research.google.com/github/seifeldinshadyy/flyrankai-Run-the-Starter-Notebooks-first-week-1-/blob/main/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ML-02 — Research Question and Provisional Lane

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

# My Lane

**Selected Lane:** Refresh / Content Opportunity Scoring

I selected the Refresh / Content Opportunity Scoring lane because it focuses on identifying content pages that should be reviewed first for potential refresh or optimization. This lane directly aligns with the work completed in the starter notebooks, where I analyzed search performance signals and built models to identify declining pages.

My goal is to create a ranking system that helps prioritize pages for review based on observable search and engagement signals rather than relying on fixed rules alone.


## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

# Research Question

### Decision

Which content pages should an SEO editor review first for a possible content refresh?

### Who takes the action?

SEO editors or content managers.

### Action

Review the recommended pages and update their content, metadata, or SEO optimization.

### Unit of Analysis

One content page.

### Output

A ranked list of pages ordered by refresh priority.

### Cost of a Wrong Recommendation

A false positive wastes editor time reviewing pages that do not require updates.

A false negative may delay improvements for pages that are actually losing search performance.

### Why ML Helps

A simple rule based on one metric (such as impressions or CTR) may miss important interactions between multiple signals. Machine learning can combine several historical signals to prioritize pages more effectively.

### Task Type

Ranking / Scoring

### Success Metric

Precision@K

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [2]:
import pandas as pd

df = pd.read_csv("/content/sample_data/content_refresh_anonymized.csv")

print("Rows:", len(df))
print("Columns:", len(df.columns))
print("Clients:", df["client_id"].nunique())
print("Average CTR (%):", round(df["ctr"].mean(),3))
print("Average Position:", round(df["avg_position"].replace(0, float("nan")).mean(),2))

Rows: 30000
Columns: 44
Clients: 32
Average CTR (%): 0.511
Average Position: 17.03


# Quick Look at the Data

I explored the starter dataset to better understand the scale of the problem before choosing my research lane.

### Key Statistics

- Number of pages: **30,000**
- Number of features: **44**
- Number of clients: **32**
- Average CTR: **0.511**
- Average Search Position: **17.03**

### Observations

The dataset contains 30,000 content pages collected from 32 anonymized clients, providing a diverse set of search performance data.

The average CTR is approximately 0.511, while the average search position is around 17.03. These metrics suggest that many pages have measurable search visibility and potential opportunities for optimization.

These statistics support my decision to focus on the **Refresh / Content Opportunity Scoring** lane because ranking pages for review based on search performance signals can help prioritize content that may benefit from updates.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

## 4. Careful words: what I can and can't claim

This project is designed to support SEO decision-making using historical search-performance data.

The goal is not to predict Google's algorithm or guarantee future rankings. Instead, the model will provide a ranked list of pages that may benefit from a content refresh based on observed patterns in the available data.

Any conclusions will be presented as observational and decision-support findings rather than causal claims.

## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.

## 5. Self-check

- [x] I chose a predefined lane.
- [x] I clearly described the decision this project improves.
- [x] I identified who will use the output and what action they will take.
- [x] I explained the cost of a wrong recommendation.
- [x] I included real numbers from the starter dataset.
- [x] I explained why machine learning may help instead of using only a simple rule.
- [x] I used careful, non-causal language.